pkg/controller/azure/node/predicate.go

```diff
@@ -110,3 +112,17 @@ func NewNodePredicate(logger logr.Logger) predicate.Predicate {
 		},
 	}
 }
+
+// nodeProjection captures only the essential properties of a node that is being cached.
+// By using projections, we prevent the cache from getting too big for clusters with large number of nodes.
+type nodeProjection struct {
```

Comment:
I also thought about proposing this, but hesitated. 
Now, that I see it, I like it very much. This actually comes close to what kube controllers do with their state of world models.

Reply:
Thanks! I will merge this then and release a new remedy-controller version.